src/Microsoft.Health.Dicom.Api/Features/Context/DicomRequestContextMiddleware.cs

```diff
@@ -51,8 +57,24 @@ public async Task Invoke(HttpContext context, IDicomRequestContextAccessor dicom
 
             dicomRequestContextAccessor.RequestContext = dicomRequestContext;
 
-            // Call the next delegate/middleware in the pipeline
-            await _next(context);
+            ByteCountingStream byteCountingStream = _responseLogStreamFactory.CreateByteCountingResponseLogStream(context.Response.Body);
+            context.Response.Body = byteCountingStream;
+
+            try
+            {
+                // Call the next delegate/middleware in the pipeline
+                await _next(context);
+            }
+            finally
+            {
+                long responseBodySize = byteCountingStream.WrittenByteCount;
+                long responseHeaderSize = HeaderUtility.GetTotalHeaderLength(context.Response.Headers);
+                long totalResponseSize = responseBodySize + responseHeaderSize;
+                context.Items["ResponseSize"] = totalResponseSize;
+
+                context.Items["TranscodedSize"] = dicomRequestContextAccessor.RequestContext.BytesTranscoded;
+                context.Items["IsTranscodeRequested"] = dicomRequestContextAccessor.RequestContext.IsTranscodeRequested;
```

Comment:
Any particular reason to put these in a property bag vs the `dicomRequestContext`?

Reply:
I thought about using the dicomRequestContext, but I left it directly on context only so I wouldn't need to pull in dicomRequestContext in workspaceplatform. I could go either way though

Reply:
Kinda depends on how it gets coded on the consuming side, but if it's just additional middleware in the same program, then it already has access to the `DicomRequestContext` through the `DicomRequestContextAccessor`

Reply:
Makes sense. BytesTranscoded and IsTranscodeRequest are already on there too so I can see how using DicomRequestContext is better. I've updated the code